api/http/security/bouncer.go

```diff
@@ -244,6 +304,26 @@ func ExtractBearerToken(r *http.Request) (string, error) {
 	return token, nil
 }
 
+// extractAPIKey extracts the api key from the api key request header or query params.
+func extractAPIKey(r *http.Request) (apikey string, ok bool) {
+	// extract the API key from the request header
+	apikey = r.Header.Get(apiKeyHeader)
+	if apikey != "" {
+		return apikey, true
+	}
+
+	// extract the API key from query params.
+	// Case-insensitive check for the "X-API-KEY" query param.
+	query := r.URL.Query()
+	for k, v := range query {
+		if strings.EqualFold(k, apiKeyHeader) {
+			return v[0], true
+		}
+	}
```

Comment:
Thinking about it some more.  API keys should not be allowed over http in my opinion as they are easily spoofed.

But particularly in the case of HTTP as those queries are often read on the wire, could be logged by a proxy or anything like that.   

https://security.stackexchange.com/questions/194108/is-it-normal-for-a-website-to-show-an-api-key-in-plain-text-that-allows-full-acc/194112

Reply:
The same issue exists with JWT> For some requests (specifically websocket ones) - we allow the jwt to be defined as query param for auth. Here we are trying to retain parity - so that API keys work in the same way.

I agree that the approach is not the most secure. It should be another point of discussion i guess (i.e. drop support for jwt or apikeys in query params).

Reply:
As discussed, query params and headers are sent in plain on `http` connection - while they are encrypted - when sent over an `https` connection.
Hence the security of these headers/data is the responsibility of portainer (i.e. if security is an important factor - then please run portainer with/on `https`)

Reply:
on a sidenote; we allow jwt auth to be defined in query params (especially for websocket connections) - we allow apikeys to also be defined in query params - simply to retain feature parity with JWT auth.